### PR TITLE
[#167615527] routing and first components

### PR DESCRIPTION
**Pull request by d-al-ibm**
added logos exported from Sketch and mock image for missing images
added Home page (complete) and initial commit for LoadingScreen and DefaultContainer
added custom css and changed import of bootstrap italia css to override coreui css
added routing in App.tsx
added customDeclarations.d.ts to fix error when importing images in ts files
added Webfont to load Titillium font
added imgMocks.ts to mock images for tests

### CODE REVIEW

File: imgMock.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = {};
```
Comment (cloudify) [inline imgMock.ts]: what's this and why it's in the project's root?

File: jest.config.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ module.exports = {
   "snapshotSerializers": ["enzyme-to-json/serializer"],
   "setupFilesAfterEnv": ["<rootDir>/setupEnzyme.ts"],
   "moduleNameMapper": {
-    ".+\\.(css|scss)$": "<rootDir>/styleMock.ts"
+    ".+\\.(css|scss)$": "<rootDir>/styleMock.ts",
```
Comment (cloudify) [inline jest.config.js]: to avoid polluting the project's root, move these mock modules to a subdirectory

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>@types dependencies were added to package.json, as a dependency for others.<br/><i>You need to move &#64;types/webfontloader into "devDependencies"?</i></td>
    </tr>
  </tbody>
</table>




## Affected stories

  * 🌟 [#167615527](https://www.pivotaltracker.com/story/show/167615527): Creazione routing e primi componenti


New dependencies added: <a href='https://www.npmjs.com/package/@types/webfontloader'><code>&#64;types/webfontloader</code></a> and <a href='https://www.npmjs.com/package/webfontloader'><code>webfontloader</code></a>.


<h2><a href="http://npmjs.com/package/@types/webfontloader">&#64;types/webfontloader</a></h2>
<p>Author: Unknown</p>
<p>Description: TypeScript definitions for typekit-webfontloader</p>
<p>Homepage: <a href="http://npmjs.com/package/@types/webfontloader">http://npmjs.com/package/@types/webfontloader</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>about 3 years ago</td></tr><tr><td>Last Updated</td><td>2 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>13</td></tr><tr><td>Direct Dependencies</td><td></td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# Installation
> `npm install --save @types/webfontloader`

# Summary
This package contains type definitions for typekit-webfontloader (https://github.com/typekit/webfontloader).

# Details
Files were exported from https://www.github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/webfontloader

Additional Details
 * Last updated: Mon, 21 Aug 2017 22:03:22 GMT
 * Dependencies: none
 * Global values: WebFont

# Credits
These definitions were written by doskallemaskin <https://github.com/doskallemaskin>.


</details>




<h2><a href="https://github.com/typekit/webfontloader">webfontloader</a></h2>
<p>Author: Unknown</p>
<p>Description: Web Font Loader gives you added control when using linked fonts via @font-face.</p>
<p>Homepage: <a href="https://github.com/typekit/webfontloader">https://github.com/typekit/webfontloader</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 4 years ago</td></tr><tr><td>Last Updated</td><td>5 months ago</td></tr><tr><td>License</td><td>Apache-2.0</td></tr><tr><td>Maintainers</td><td>2</td></tr><tr><td>Releases</td><td>34</td></tr><tr><td>Keywords</td><td>web, fonts, webfonts, font, loader and @font-face</td></tr>
</table>
This README is too long to show.

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2d584971cfc1d969dd9878839b855493d0a26959
</p>

File: src/App.tsx [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
 export const App: React.FC = () => {
   return (
-    <div className="App">
-      <h1 className="primary-bg white-color">Piattaforma Onboard PA</h1>
-    </div>
+    <BrowserRouter>
+      <div className="App h-100">
+        <Route path="/" exact component={Home} />
+        <Route
+          path="/(generazione-documenti|conferma-documenti)"
```
Comment (cloudify) [inline src/App.tsx]: use english for paths

File: src/containers/DefaultContainer/DefaultContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import * as React from "react";
+
+export const DefaultContainer: React.FC = () => {
```
Comment (cloudify) [inline src/containers/DefaultContainer/DefaultContainer.tsx]: describe the purpose of this component

File: src/containers/Home/Home.css [hunk 5]
```diff
@@ -0,0 +1,50 @@
+.Home {
+    text-align: center;
+}
+.Home .custom-background-container {
+    background-image: linear-gradient(270deg, #0066CC 26%, #0C00D3 78%);
+}
+
+.Home .io-icon-home {
```
Comment (cloudify) [inline src/containers/Home/Home.css]: @matteodesanti are you ok with all these hardcoded pixel values?

File: src/containers/Home/Home.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
+import * as React from "react";
+
+import ioLogoWhite from "../../assets/img/io-logo-white.svg";
+import mockImage from "../../assets/img/mock-img-white.png";
+import "./Home.css";
+
+export const Home: React.FC = () => {
```
Comment (cloudify) [inline src/containers/Home/Home.tsx]: describe the purpose of this component

File: src/containers/Home/Home.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+              className="btn btn-primary home-btn home-btn-mt"
+            >
+              Responsabile legale ente
+            </button>
+          </div>
+        </div>
+        <div className="row justify-content-start home-row-logos">
+          <div className="col-6 text-right">
+            {/*TODO: fix with right images when available - keep track with story 167623550*/}
+            <img src={mockImage} width={220} />
```
Comment (cloudify) [inline src/containers/Home/Home.tsx]: btw you could use something like lorem picsum here

File: src/containers/LoadingScreen/LoadingScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import * as React from "react";
+
+export const LoadingScreen: React.FC = () => {
```
Comment (cloudify) [inline src/containers/LoadingScreen/LoadingScreen.tsx]: describe the purpose of this component

File: src/customDeclarations.d.ts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+declare module "*.png";
```
Comment (cloudify) [inline src/customDeclarations.d.ts]: keep ambient declaration in dedicated directory (e.g. `src/@types`)

File: src/customDeclarations.d.ts [hunk 10]
```diff
@@ -0,0 +1,2 @@
...
+declare module "*.svg";
```
Comment (cloudify) [inline src/customDeclarations.d.ts]: also each module should be in its own declaration file - in this case you can have one for png and one for svg

File: src/index.tsx [hunk 11]
```diff
@@ -1,5 +1,13 @@
 import * as React from "react";
 import { render } from "react-dom";
+import WebFont from "webfontloader";
+
 import { App } from "./App";
 
+WebFont.load({
```
Comment (cloudify) [inline src/index.tsx]: we can't use google webfonts (privacy issues), we should host the webfont with the app

File: package.json [hunk 12]
```diff
@@ -44,6 +44,7 @@
     "@coreui/coreui-plugin-chartjs-custom-tooltips": "^1.3.1",
     "@coreui/icons": "0.3.0",
     "@coreui/react": "^2.5.1",
+    "@types/webfontloader": "^1.6.29",
```
Comment (cloudify) [inline package.json]: types packages should go under `devDependencies` (see [Danger error](https://github.com/teamdigitale/io-onboarding-pa-fe/pull/2#issuecomment-520406194))

File: src/components/Home/Home.tsx [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+import logoHomeDel from "../../assets/img/logo_home_del.png";
+import logoHomeDev from "../../assets/img/logo_home_dev.png";
+import logoHomeRep from "../../assets/img/logo_home_legal_rep.png";
+
+import "./Home.css";
+
+/*Landing page with three different login for developers, institution delegate and institution legal ref*/
+export const Home = () => {
+  /*array containing three login elements props*/
+  const homeLoginButtonsArray: ReadonlyArray<{
```
Comment (cloudify) [inline src/components/Home/Home.tsx]:
```suggestion
  const homeLoginButtonsArray: ReadonlyArray<React.ComponentProps<typeof HomeLoginButton>>
```

File: src/components/Home/HomeLoginButton/HomeLoginButton.tsx [hunk 14]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Button, Col, Media, Row } from "reactstrap";
+import ioLogoWhite from "../../../assets/img/io-logo-white.svg";
+
+/*props for LoginHomeButton component*/
```
Comment (cloudify) [inline src/components/Home/HomeLoginButton/HomeLoginButton.tsx]: use block comments for classes, interfaces, functions

```ts
/**
 * ...
 */
```

File: src/components/Home/HomeLoginButton/HomeLoginButton.tsx [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+  buttonText: string;
+  img: string;
+  imgHeight: number;
+  imgWidth: number;
+  link?: string;
+  offset?: string;
+  text: string;
+}
+
+/*component defining a login element used in Home component(image, button and descriptive text)*/
```
Comment (cloudify) [inline src/components/Home/HomeLoginButton/HomeLoginButton.tsx]: use block comments

File: src/components/LoadingScreen/LoadingScreen.tsx [hunk 16]
```diff
@@ -0,0 +1,6 @@
+import * as React from "react";
+
+/*Component that returns loading and informative screens across application */
```
Comment (cloudify) [inline src/components/LoadingScreen/LoadingScreen.tsx]: use block comments

**Pull request merged**